scripts/comparator/utils/compare_utilities.go

```diff
@@ -0,0 +1,203 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"go.uber.org/zap"
+)
+
+const queryFmt string = "/api/v1/query_range?start=%d&end=%d&query=%s&step=%s"
+
+// InputQueries is a slice of InputQuery.
+type InputQueries []InputQuery
+
+// InputQuery is the JSON representation of a query to be compared.
+type InputQuery struct {
+	// QueryGroup is the general category for these queries.
+	QueryGroup string `json:"queryGroup"`
+	// Queries is the list of raw queries.
+	Queries []string `json:"queries"`
+	// Steps is the list of step sizes for these queries.
+	Steps []string `json:"steps"`
+}
+
+// PromQLQueryGroup is a list of constructed PromQL query groups.
+type PromQLQueryGroup struct {
+	// QueryGroup is the general category for these queries.
+	QueryGroup string
+	// Queries is a list of PromQL compatible queries.
+	Queries []string
+}
+
+func (q InputQueries) constructPromQL(
+	start int64,
+	end int64,
+) []PromQLQueryGroup {
+	queries := make([]PromQLQueryGroup, 0, len(q))
+	for _, inQuery := range q {
+		queries = append(queries, inQuery.constructPromQL(start, end))
+	}
+
+	return queries
+}
+
+func (q InputQuery) constructPromQL(start int64, end int64) PromQLQueryGroup {
+	queries := make([]string, 0, len(q.Queries)*len(q.Steps))
+	for _, inQuery := range q.Queries {
+		for _, inStep := range q.Steps {
+			q := strings.ReplaceAll(inQuery, " ", "+")
+			query := fmt.Sprintf(queryFmt, start, end, q, inStep)
+			queries = append(queries, query)
+		}
+	}
+
+	return PromQLQueryGroup{
+		QueryGroup: q.QueryGroup,
+		Queries:    queries,
+	}
+}
+
+func parseFileToQueries(
+	fileName string,
+	log *zap.Logger,
+) (InputQueries, error) {
+	file, err := os.Open(fileName)
+	if err != nil {
+		log.Error("could not open file", zap.Error(err))
+		return nil, err
+	}
+
+	defer file.Close()
+	buf, err := ioutil.ReadAll(file)
+	if err != nil {
+		log.Error("could not read file", zap.Error(err))
+		return nil, err
+	}
+
+	queries := make(InputQueries, 10)
```

Comment:
nit: Perhaps just make this capacity not size uup front allocated?  `make(foo, 0, 10)`? I guess wouldn't make a difference here (not sure how json.Unmarshal works with pre-existing slices) but might be good hygiene?

Reply:
Sounds good, yeah

Reply:
Yep, sounds good